kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/utils/TopicToTableResolver.java

```diff
@@ -34,75 +37,79 @@
-                    + " regexes for both '" + previousPattern
-                    + "' and '" + secondMatch + "'"
+            throw new ConfigException("Value '" + value
+              + "' for property '" + BigQuerySinkConfig.TOPICS_CONFIG
+              + "' matches " + BigQuerySinkConfig.TOPICS_TO_TABLES_CONFIG
+              + " regexes for both '" + previousPattern
+              + "' and '" + secondMatch + "'"
             );
           }
           String formatString = pattern.getValue();
           try {
             match = patternMatcher.replaceAll(formatString);
             previousPattern = pattern.getKey().toString();
-          } catch (IndexOutOfBoundsException e) {
-            throw new ConfigException(
-                "Format string '" + formatString
-                    + "' is invalid in property '" + patternProperty
-                    + "'", e);
+          } catch (IndexOutOfBoundsException err) {
+            throw new ConfigException("Format string '" + formatString
+              + "' is invalid in property '" + BigQuerySinkConfig.TOPICS_TO_TABLES_CONFIG
+              + "'", err);
           }
         }
       }
       if (match == null) {
         match = (sanitize) ? sanitizeTableName(value) : value;
       }
-      matches.put(value, match);
+      String dataset = topicsToDatasets.get(value);
+      matches.put(value, TableId.of(dataset, match));
     }
     return matches;
   }
 
-  private static String sanitizeTableName(String tableName) {
-    // Take anything that isn't valid in a table name and turn it into an underscore.
+  /**
+   * Return a Map detailing which topic each table corresponds to. If sanitization has been enabled,
+   * there is a possibility that there are multiple possible schemas a table could correspond to. In
+   * that case, each table must only be written to by one topic, or an exception is thrown.
+   *
+   * @param config Config that contains properties used to generate the map
+   * @return The resulting Map from TableId to topic name.
+   */
+  public static Map<TableId, String> getTablesToTopics(BigQuerySinkConfig config) {
+    Map<String, TableId> topicsToTableIds = getTopicsToTables(config);
+    Map<TableId, String> tableIdsToTopics = new HashMap<>();
+    for (Map.Entry<String, TableId> topicToTableId : topicsToTableIds.entrySet()) {
+      if (tableIdsToTopics.put(topicToTableId.getValue(), topicToTableId.getKey()) != null) {
+        throw new ConfigException("Cannot have multiple topics writing to the same table");
+      }
+    }
+    return tableIdsToTopics;
+  }
+
+  /**
+   * Strips illegal characters from a table name. BigQuery only allows alpha-numeric and
+   * underscore. Everything illegal is converted to an underscore.
+   *
+   * @param tableName The table name to sanitize.
+   * @return A clean table name with only alpha-numerics and underscores.
+   */
+  public static String sanitizeTableName(String tableName) {
```

Comment:
Just seemed generally useful. I can swap back to private, though. Nothing else is using it.